File: Rasha_Toine/Rasha/QLForm/src/ql/ast/expression/Not.java
package ql.ast.expression;

import ql.visiting.ExpressionVisitor;

public class Not extends UnaryExpression {

	public Not(Expression expr) {
		super(expr);
	}
	
	@Override
	public <T, U> T accept(ExpressionVisitor<T, U> visitor, U ctx) {
		return visitor.visit(this, ctx);
	}
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QL/AST/Expression/Not.cs
namespace Assignment1.Model.QL.AST.Expression
{
    public class Not : ASTNode, IExpression
    {
        public IExpression Expression { get; }

        public Not(IExpression expression)
        {
            Expression = expression;
        }

        public Not(int lineNumber, IExpression expression)
        {
            _lineNumber = lineNumber;
            Expression = expression;
        }

        public void Accept(IExpressionVisitor visitor) => visitor.Visit(this);
    }
}
